common/app/common/commercial/EditionAdTargeting.scala

```diff
@@ -1,42 +1,83 @@
 package common.commercial
 
-import com.gu.commercial.display.{AdCall, AdCallParamKey}
+import com.gu.commercial.display._
 import com.gu.contentapi.client.model.v1.{Content, Section, Tag}
 import common.Edition
 import ophan.SurgingContentAgent
-import play.api.libs.json.Json
+import play.api.libs.json._
 
-case class EditionAdTargeting(edition: Edition, params: Map[String, String])
+case class EditionAdTargeting(edition: Edition, params: Map[String, Either[String, Set[String]]])
 
 object EditionAdTargeting {
 
+  implicit val paramsWrites = new Writes[Either[String, Set[String]]] {
```

Comment:
What's the value of having a `Either` rather than a simple ADT (with more meaningful name)?